lib/action_cable/connection/base.rb

```diff
@@ -165,6 +165,28 @@ def on_close
         end
 
 
+        def allow_request_origin?
+          return true if server.config.disable_request_forgery_protection
+
+          if env['HTTP_ORIGIN'].present?
+            origin_host = URI.parse(env['HTTP_ORIGIN']).host
+
+            allowed = if server.config.allowed_request_origins.present?
+              Array.wrap(server.config.allowed_request_origins).include? origin_host
+            else
+              request.host == origin_host
```

Comment:
How would the Host and the Origin match for a cross site hijacking though? I think we should have some sort of default at least.